docs/DeveloperGuide.md

```diff
@@ -1,34 +1,1135 @@
+
+**Implementation**  
+When the user types `clear` the following sequence occurs. 
+1. The user keys in `clear`.
+    
+    1. A `DietSessionUi` component will call `dietSessionUI.getInput()`. 
+    1. Input will be parsed in `processCommand()`.   
+    
+2. Creation of command object from input
+    1. This will create a FoodItemClear() instantiation of which the method execute() is called.
+    
+3. Executing Command
+    1. The ArrayList Clear method is called and removes all Food entries from the ArrayList.
+
+#### 4.3.2.6. Stopping the recording of diet session data: `end`
+
+The feature allows users to end the current diet session and return back to the diet manager.
+
+**Implementation**  
+When the user types `end` the following sequence occurs. 
+1. The user keys in `end`.
+    
+    1. A `DietSessionUi` component will call `dietSessionUI.getInput()`. 
+    1. Input will be parsed in `processCommand()`.   
+    
+2. Exiting of inputLoop()
+    The inputLoop() exits when userInput.equals("end").
+
+#### 4.3.3. List all past diet sessions: `list`
+
+The feature allows users to view all past created diet sessions.
+
+**Implementation**  
+When the user types `list` in a diet manager instance the following sequence occurs. 
+1. The user keys in `list`.
+    
+    1. A `DietSessionUi` component will call `dietSessionUI.getInput()`. 
+    1. Input will be parsed in `processCommand()`.   
+    
+2. Creation of command object from input
+    1. This will create a FoodItemList() instantiation of which the method execute() is called.
+    
+3. Executing Command
+    1. A for loop iterates through the entire ArrayList<Food> and prints out every item with their calories.
+
+The sequence diagram below summarizes how listing past Diet sessions work:
+
+![Load Data Sequence Diagram](pictures/Zeon/DietSessionList.png)
+#### 4.3.4. Edit a past diet session: `edit`
+
+The feature allows users to edit previously created diet sessions.
+
+**Implementation**  
+When the user types `edit [INDEX_OF_SESSION]` the following sequence occurs. 
+1. The user keys in `edit 1`.
+    
+    1. A `DietSessionUi` component will call `dietSessionUI.getInput()`. 
+    1. Input will be parsed in `processCommand()`.   
+    
+2. Creation of `DietSessionEdit` command object from input
+    1. A `DietSessionEdit()` command class instantiation is created and the execute() method is called.
+    
+3. Executing Command
+    1. This will call `readDietSession()` from `storage.diet.DietStorage` and it reads the file stored at `saves/diet`.
+    1. The `start()` method is then called in the diet session, starting a diet session instance.
+    
+4. After Execution
+    1. The diet session instance is then saved by calling `writeToStorageDietSession()` from `DietStorage`
+
+The sequence diagram below summarizes how editing Diet session works:
+
+![Load Data Sequence Diagram](pictures/Zeon/DietSessionEdit.png)
```

Comment:
![CS2113(4)](https://user-images.githubusercontent.com/65339123/97545868-96f9a080-1a06-11eb-915a-444f8bfe8eef.PNG)

I think the activation bar should stop after reading diet session and continue when writing.